modules/backup-azure/client.go

```diff
@@ -0,0 +1,177 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modstgazure
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+	"github.com/weaviate/weaviate/entities/backup"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+)
+
+type azureClient struct {
+	client     *azblob.Client
+	config     clientConfig
+	serviceURL string
+	dataPath   string
+}
+
+func newClient(ctx context.Context, config *clientConfig, dataPath string) (*azureClient, error) {
+	connectionString := os.Getenv("AZURE_STORAGE_CONNECTION_STRING")
```

Comment:
I see a few env vars with `AZURE_STORAGE_*` prefixes. Is the `STORAGE` part an Azure naming convention, or something that is being added on by us?

Reply:
Those `AZURE_STORAGE_*` vars are the ones that come from `Azure` (like `AWS_*` vars for `AWS`)
The most important ones are: `AZURE_STORAGE_ACCOUNT_NAME` and `AZURE_STORAGE_PRIMARY_ACCOUNT_KEY` with those 2 (or only account name in case of anonymous access) you are able to authenticate agains `Azure Storage` service. You can also pass `AZURE_STORAGE_CONNECTION_STRING` but from I read this is mostly used for tests, but can be used to authenticate.

Reply:
I was thinking we could drop the `STORAGE` part of the name, but maybe we will have other `AZURE` env vars in the future which will be unrelated to storage?

Reply:
hmm I took those env variable names from their official [azure-go-sdk](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob) example

Reply:
I will check what other Azure env variables are there

Reply:
I have used `AZURE_STORAGE_ACCOUNT` `AZURE_STORAGE_KEY` and `AZURE_STORAGE_CONNECTION_STRING` which standard ones, mentioned in Azure docs